pyacq/core/sharedarray.py

```diff
@@ -55,13 +43,62 @@ def __init__(self, shape=(1,), dtype = 'float64', shm_id = None):
                 self.mmap = mmap.mmap(self._tmpFile.fileno(), self.nbytes, mmap.MAP_SHARED, mmap.PROT_READ)
                 
     def close(self):
+        """Close this buffer.
+        """
         self.mmap.close()
         if not sys.platform.startswith('win') and hasattr(self, '_tmpFile'):
             self._tmpFile.close()
     
     def to_dict(self):
-        return {'shape': self.shape, 'dtype': self.dtype, 'shm_id': self.shm_id}
+        """Return a dict that can be serialized and sent to other processes to
+        access this buffer.
+        """
+        return {'nbytes': self.nbytes, 'shm_id': self.shm_id}
+    
+    def to_numpy(self, offset, dtype, shape, strides):
+        """Return a numpy array pointing to part (or all) of this buffer.
+        """
+        return np.ndarray(buffer=self.mmap, shape=shape,
+                          strides=strides, offset=offset, dtype=dtype)        
+        
+
+
```

Comment:
I really like the 1D ring buffer.
But I am not so happy with the ND RingBufferArray.
double=True with timeaxis=0 internnaly for 2D array have no sense because the channel is split in 2 part in the memory and then we cannot garanty that all reading are continuous for one channel.

I think this is an advance feature. By default everything should be (channel, sample) internally and externally but for some rare case a swapaxes option is needed. For instance I want to perform lots of FFT on some channel, we really gain to have internnalu continuous channel. So to have internnaly double=True and timeaxis=1. This garanty no copy at reading but at a cost of double writting and transposing.

Could I make a proposal starting from your branch ?